lib/bundles.cf

```diff
@@ -382,6 +382,47 @@ bundle agent cmerge(name, varlist)
       "$(name)_str" string => format("%S", $(name)),    policy => "free";
 }
 
+bundle agent collect(name, tag)
+# @brief bundle to collect values from tagged variables into a data container
+# @param name the variable name to create
+# @param tag the tags to match
+#
+# The result will be in `collect.$(name)`.  You can also use
+# `cmerge.$(name)_str` for a string version of the merged containers.
+#
+# The name is variable so you can call this bundle for more than one
+# collection.
+#
+# If you merge a key-value map into an array or vice versa, the map
+# always wins.  So this example will result in a key-value map even
+# though `cmerge.$(name)` starts as an array.
+#
+# **Example:**
+#
+# ```cf3
+# bundle agent run
+# {
+#   vars:
+#       # the "mymerge" tag is user-defined
+#       "a"       data  => parsejson('{ "mark": "b" }'), meta => { "mymerge" };
+#       "b"       data  => parsejson('{ "volker": "h" }'), meta => { "mymerge" };
```

Comment:
Can we come up with a better example that is more sysadmin oriented? Maybe we ports that should be open?
If not, then lets make the names more specific: for example instead of b and h use the full last names, and instead of collect.all maybe collect.last_names?